sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -0,0 +1,147 @@
+      await this.studentLoanBalancesIntegrationService.getResponseFilesFullPath(
+        this.esdcConfig.ftpResponseFolder,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.PT.MBAL.[0-9]{8}.*`,
+          "i",
+        ),
+      );
+    for (const remoteFilePath of remoteFilePaths) {
+      const serviceProcessSummary = new ProcessSummary();
+      parentProcessSummary.children(serviceProcessSummary);
+      await this.processFile(
+        remoteFilePath,
+        serviceProcessSummary,
+        auditUserId,
+      );
+    }
+  }
+
+  /**
+   * Process each individual Student Loan Balances response file from the SFTP.
+   * @param remoteFilePath Student Loan Balances response file to be processed.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   */
+  private async processFile(
+    remoteFilePath: string,
+    childrenProcessSummary: ProcessSummary,
+    auditUserId: number,
+  ): Promise<void> {
+    childrenProcessSummary.info(`Processing file ${remoteFilePath}.`);
+    let studentLoanBalancesSFTPResponseFile: StudentLoanBalancesSFTPResponseFile;
+    try {
+      studentLoanBalancesSFTPResponseFile =
+        await this.studentLoanBalancesIntegrationService.downloadResponseFile(
+          remoteFilePath,
+        );
+    } catch (error) {
+      this.logger.error(error);
+      childrenProcessSummary.warn(
+        `Error downloading file ${remoteFilePath}. Error: ${error}`,
+      );
+    }
+    childrenProcessSummary.info(
+      `File contains ${studentLoanBalancesSFTPResponseFile.records.length} Student Loan balances.`,
+    );
+    // Get only the file name for logging.
+    const fileName = path.basename(remoteFilePath);
+    try {
+      await this.dataSource.transaction(async (transactionalEntityManager) => {
+        const studentLoanBalancesRepo =
+          transactionalEntityManager.getRepository(StudentLoanBalance);
+        for (const studentLoanBalanceRecord of studentLoanBalancesSFTPResponseFile.records) {
+          const student = await this.studentService.getStudentByPersonalInfo(
+            studentLoanBalanceRecord.sin,
+            studentLoanBalanceRecord.lastName,
+            getISODateOnlyString(studentLoanBalanceRecord.birthDate),
+            transactionalEntityManager,
+          );
+          // If student not found continue.
+          if (!student) {
+            childrenProcessSummary.error(
+              `Student not found for line ${studentLoanBalanceRecord.lineNumber}.`,
+            );
+            continue;
+          }
+          const auditUser = { id: auditUserId } as User;
+          const now = new Date();
+          await studentLoanBalancesRepo.insert({
+            student: student,
+            cslBalance: studentLoanBalanceRecord.cslBalance,
+            balanceDate: getISODateOnlyString(
+              studentLoanBalancesSFTPResponseFile.header.balanceDate,
+            ),
+            createdAt: now,
```

Comment:
Just as a suggestion, we explicitly set `now` when it will be used in multiple places and it would be good to have those multiple places in sync.